### PR TITLE
Misc nRF54H20 updates

### PR DESCRIPTION
**Pull request by 57300**
Alternative to https://github.com/nrfconnect/sdk-zephyr/pull/1642 which tracks latest.

### CODE REVIEW

**Comment by e-rk** [comment]
@57300 would it be possible to split [[nrf fromtree] boards: nordic: nrf54h20dk: Update memory map](https://github.com/nrfconnect/sdk-zephyr/pull/1681/commits/c72c8167b503cd15f7dd287e3f8c209b1501e18e) to a separate PR to merge it sooner? It is needed by multiple people to work on nRF54H20.

**Review by gmarull — APPROVED** [review]

**Review by anangl — APPROVED** [review]

File: modules/hal_nordic/CMakeLists.txt [hunk 1]
```diff
@@ -11,7 +11,7 @@ if(CONFIG_NRF_REGTOOL_GENERATE_UICR)
   list(APPEND nrf_regtool_components GENERATE:UICR)
 endif()
 if(DEFINED nrf_regtool_components)
-  find_package(nrf-regtool 5.1.0 REQUIRED
+  find_package(nrf-regtool 5.1.0
```
Comment (gmarull) [inline modules/hal_nordic/CMakeLists.txt:14]: oops, this shouldn't be undone

**Review by gmarull — CHANGES_REQUESTED** [review]

File: modules/hal_nordic/CMakeLists.txt (hunk 1 above)
Comment (57300) [reply modules/hal_nordic/CMakeLists.txt]: Undid the undo.

File: dts/bindings/spi/nordic,nrf-exmif.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# Copyright (c) 2024 Nordic Semiconductor ASA
```
Comment (e-rk) [inline dts/bindings/spi/nordic,nrf-exmif.yaml]: Why is this commit in this PR? This change was already merged to sdk-zephyr main: https://github.com/nrfconnect/sdk-zephyr/commit/7ea6937f02affb74cae2327aeca955ae5e5c3106

Comment (57300) [reply dts/bindings/spi/nordic,nrf-exmif.yaml]: The EXMIF patch is reverted and re-applied to avoid conflicts.

**Review by gmarull — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**